internal/cli/cli.go

```diff
@@ -66,14 +69,49 @@ func collectionCmdFetch(rootCmd *cobra.Command) *cobra.Command {
 			if err != nil {
 				return fmt.Errorf("required flag '%s' not set: %w", flagNameApiKey, err)
 			}
+
+			client := outline.New(baseUrl, &http.Client{}, apiKey)
+			for _, colId := range args {
+				st, err := client.Collections().Structure(outline.CollectionID(colId)).Do(context.Background())
+				if err != nil {
+					return fmt.Errorf("can't get collection with id '%s': %w", colId, err)
+				}
+
+				b, err := json.MarshalIndent(st, "", "  ")
+				if err != nil {
+					return fmt.Errorf("failed marshalling collection with id '%s: %w", colId, err)
+				}
+				fmt.Println(string(b))
+			}
+			return nil
+		},
+	}
+}
+
+func collectionCmdInfo(rootCmd *cobra.Command) *cobra.Command {
+	return &cobra.Command{
+		Use:   "info",
+		Short: "Get collection info",
+		Long:  "Get information about the collection",
+		Args:  cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			baseUrl, err := rootCmd.Flags().GetString(flagNameBaseUrl)
+			if err != nil {
+				return fmt.Errorf("required flag '%s' not set: %w", flagNameBaseUrl, err)
+			}
+			apiKey, err := rootCmd.Flags().GetString(flagNameApiKey)
+			if err != nil {
+				return fmt.Errorf("required flag '%s' not set: %w", flagNameApiKey, err)
+			}
```

Comment:
We dublicate this already three times.
Maybe it make sense to introduce a function like
```go
func requiredFlags(rootCmd *cobra.Command) (baseUrl string, apiKey string, err error) {
...
}
```

An alternative would be to move the `outline` client creation outside of these sub command to its "parent" command (the `collection` command).
Then we check the required flag only once. If successfully, we can create the `outline` client and put it as an argument to the sub-commands...

What do you think? 🤔

Reply:
Yes, but doing this properly requires enough refactoring to make it worthy of a separate task.